test/unit/validators/subscriptionValidator.test.js

```diff
@@ -0,0 +1,50 @@
+const {expect} = require("chai");
+const SubscriptionValidator = require('../../../src/domain/validators/subscriptionValidator');
+const CityValidatorMock = require('../../mocks/validators/cityValidator.mock');
+
+const cityValidatorMock = new CityValidatorMock();
+const subscriptionValidator = new SubscriptionValidator(cityValidatorMock);
+
+const validSub = {
+    email: 'valid@mail.com',
+    city: 'Kyiv',
+    frequency: 'hourly'
+}
+
+describe('SubscriptionValidator Unit Tests', () => {
+
+    it('should return true for success for valid subscription data', async () => {
+        const result = await subscriptionValidator.validateNewSubscription(validSub.email, validSub.city, validSub.frequency);
+        expect(result).to.deep.eq({ success: true, err: '' });
+    });
+
+    it('should return false for success and an error message for missing email', async () => {
+        const result = await subscriptionValidator.validateNewSubscription(null, validSub.city, validSub.frequency);
+        expect(result).to.deep.eq({ success: false, err: 'MISSING REQUIRED FIELDS' });
+    });
+
```

Comment:
Additionally, consider defining expected error messages as constants to avoid hardcoding them in multiple tests.
```suggestion
const ERRORS = {
  MISSING_REQUIRED_FIELDS: 'MISSING REQUIRED FIELDS',
  INVALID_EMAIL_FORMAT: 'INVALID EMAIL FORMAT',
  INVALID_FREQUENCY: 'INVALID FREQUENCY',
  INVALID_CITY: 'INVALID CITY',
};
```

Reply:
As each error is only referenced once in the test, i think keeping these as strings is ok for now. There is a whole question of handling data transferring and errors across my app (eg using throw/catch vs error messages vs error objects), as now i use a unified DTO with possible weather or subscription data fields. Perhaps later global error objects will be introduced as constants, and i'll refine all the tests to use them instead of string messages.